### ISSUE TITLE
Spark dependency needs to be re-evaluated

### ISSUE THREAD

**Issue by rsreds** [SimoneGasperini/cms-cmSim#3]
While Spark provides a nice tool to do high performance parallel data analysis it also requires a great effert in its setup and makes it difficult to contribute to the repository.

In addition Spark shows its best value in a distributed environment not in a single node setup (such as a personal machine).

Few alternatives should be evaluated, such as Pandas dataframe and numpy. If advanced parallelism is required (more than that provided by numpy) consider Dask.

**Comment by SimoneGasperini** [comment]
Yes, Pandas (and Numpy) would be the simplest (and probably most flexible) option and we will avoid any kind of Spark dependency issue.

However, even if we run the code on a personal machine and don't leverage what Spark offers in a distributed environment, there may be reasons to prefer it over others (e.g. it doesn't require loading the entire dataframes into memory, which may create problems even for not so huge input files).

I don't know if there are any tools that are more suitable for our case but probably we have to rethink.

**Comment by rsreds** [comment]
I suggest to evaluate (roughly) the time gained using spark df and when using only pandas.
If memory is an issue (we are dealing with a large dataset) try loading only necessary columns rather than entire dataset, or use chunking when possible. See [Pandas user guide](https://pandas.pydata.org/pandas-docs/stable/user_guide/scale.html).
If this proves to be problematic, we could test [Dask ](https://docs.dask.org/en/latest/)(which is python native) or stick with Spark (since [v2.2](https://issues.apache.org/jira/browse/SPARK-18267?jql=project%20%3D%20SPARK%20AND%20component%20%3D%20PySpark%20AND%20text%20~%20%22pip%22) pip will install spark standalone when run `pip install pyspark`)

**Comment by SimoneGasperini** [comment]
I did a quick test and actually Pandas should be enough for the moment, so I proceed to remove the dependency on Spark.
To efficiently create the Pandas dataframe from the .parquet file, the [PyArrow](https://arrow.apache.org/docs/python/index.html) library is needed when calling `read_parquet`. Reading the file with the analogous function in Spark and then convert the dataframe by the method `toPandas()` is much slower.

In case of extra needs for parallel computing in the future, we may consider Dask (as you suggested) which seems to provide good integration with Pandas itself.
